newIDE/app/src/Profile/ProfileDialog.js

```diff
@@ -17,33 +17,34 @@ import Window from '../Utils/Window';
 import { showErrorBox } from '../UI/Messages/MessageBox';
 import CreateProfile from './CreateProfile';
 import PlaceholderLoader from '../UI/PlaceholderLoader';
-import { type GameDetailsTab } from '../GameDashboard/GameDetailsDialog';
+import RouterContext from '../MainFrame/RouterContext';
 
 export type ProfileTab = 'profile' | 'games-dashboard';
 
 type Props = {|
   currentProject: ?gdProject,
   open: boolean,
   onClose: () => void,
-  initialTab: ProfileTab,
-  gamesDashboardInitialGameId: ?string,
-  gamesDashboardInitialTab: ?GameDetailsTab,
-  onGameDetailsDialogClose: () => void,
 |};
 
-const ProfileDialog = ({
-  currentProject,
-  open,
-  onClose,
-  initialTab,
-  gamesDashboardInitialGameId,
-  gamesDashboardInitialTab,
-  onGameDetailsDialogClose,
-}: Props) => {
-  const [currentTab, setCurrentTab] = React.useState<ProfileTab>(initialTab);
+const ProfileDialog = ({ currentProject, open, onClose }: Props) => {
+  const { routeArguments, removeRouteArguments } = React.useContext(
+    RouterContext
+  );
+  const [currentTab, setCurrentTab] = React.useState<ProfileTab>('profile');
   const authenticatedUser = React.useContext(AuthenticatedUserContext);
   const isUserLoading = authenticatedUser.loginState !== 'done';
 
+  React.useEffect(
+    () => {
+      if (routeArguments['initial-dialog'] === 'games-dashboard') {
+        setCurrentTab('games-dashboard');
+        removeRouteArguments(['initial-dialog']);
```

Comment:
I feel like this logic will happen often: "If arguments, then do this then remove arguments". That might become a method provided by the context

Reply:
yeah I agree and yet, I'm not sure how to simplify this for the moment.
Ideally, in the future, we actually never remove that argument so that the argument matches what's currently open